map.go

```diff
@@ -129,22 +130,25 @@ func createMap(spec *MapSpec, inner *internal.FD, handle *btf.Handle) (*Map, err
 		spec.ValueSize = 4
 
 	case PerfEventArray:
-		if spec.KeySize != 0 {
-			return nil, errors.Errorf("KeySize must be zero for perf event array")
-		}
-		if spec.ValueSize != 0 {
-			return nil, errors.Errorf("ValueSize must be zero for perf event array")
-		}
-		if spec.MaxEntries == 0 {
-			n, err := internal.OnlineCPUs()
-			if err != nil {
-				return nil, errors.Wrap(err, "perf event array")
+		if spec.KeySize != 4 || spec.ValueSize != 4 {
+			// do not validate them if they're already correct
+			if spec.KeySize != 0 {
```

Comment:
I think doing the following might make the code a bit simpler:
```suggestion
			if spec.KeySize != 0 || spec.KeySize != 4 {
```

Reply:
Hey, thanks for taking a look.

yeah i think rewriteMaps would do the trick for me without needing to add this hacky spec field if it's exposed for me to call. 

w.r.t. that keysize valuesize validation:
i need to look at the changes i made again but what it boiled down to was one of my fellow devs would do some of his quick-and-dirty testing via tc and iproute2 calls to set tc-bpf actions and xdp programs, but a reference to PerfEventArray with keysize and valuesize set to 0 makes whatever those two do for bpf loading throw errors concerning the map. i don't think it's a huge deal, it was more a matter of convenience for him since primarily we want to be loading using this stuff anyways.

anyways. if i'm able to call rewriteMaps, then this PR can just be turned down, the 0 or 4 thing was just a strange curiosity to me since it's gonna be set to 4 anyways.